### DISCUSSION TITLE
Is there an easy way to set up CI/CD for pockethost + github?

### DISCUSSION THREAD

**Discussion by snorresovold** [pockethost/pockethost#189] [Q&A]
Id like to set up CI/CD with pockethost and github, is this possible in any way

**Comment by benallfree** [comment]
@snorresovold Can you provide a little more information about what you're trying to do?

**Comment by snorresovold** [comment]
I'm trying to make it so that when i update my local pocketbase on my github repo, when i push it to a branch it updates on pockethost as well

**Reply by benallfree** [reply]
> update my local pocketbase

I'm still not understanding. PocketHost just runs instances of PocketBase for you. What are you trying to deploy to PocketHost?

If you can share a link to your repo, it will give me a better idea of what you mean.

**Reply by snorresovold** [reply]
I don't currently have pocketbase in my repo. I would like to make changes on my local pocketbase instance and then automaticcaly bring those changes to pockethost.

**Reply by benallfree** [reply]
How are you making changes to PocketBase? Do you mean you are running a custom build of PocketBase (the Go executable)? Or do you mean schema and data changes, such as adding/removing/modifying collections and records?

**Reply by snorresovold** [reply]
Schema and data changes.

**Reply by benallfree** [reply]
Thank you for clarifying.

Schema changes can be made using the admin import/export JSON feature of PocketBase. 

If you wanted to automate deployment of schema changes, you could create a Github Action that logs into your PocketBase (hosted on PocketHost) instance as an admin and makes an API call to apply the latest schema changes from your repo.

If you do decide to create that script, please make a separate repo and share it with [awesome-pocketbase](https://github.com/benallfree/awesome-pocketbase).

@ganigeorgiev What recommendations do you have for data (not schema) migrations? Is there any data import/export JSON feature planned?

**Reply by ganigeorgiev** [reply]
For schema PocketBase has [migrations](https://pocketbase.io/docs/migrations/). If you are using the prebuilt executables, every collection change via the web API (eg. from the Admin UI, manual import, etc.) will generate a js migration file located in `pb_migrations` (all migrations are executed automatically on `serve`).

For data migrations at the moment there is no automated built-in command for that and it's up to the developers to handle on their own by:
- manually creating custom migrations - https://pocketbase.io/docs/migrations/#migrate-create
- directly inserting records into the db via the sqlite3 cli tool (eg. `sqlite3 pb_data/data.db '.import test.csv foo --csv'`)
- custom scripts that call the web API(s)

Import/export of records as JSON is in the roadmap but other tasks are with higher priority and for now is "On hold" - https://github.com/pocketbase/pocketbase/issues/48#issuecomment-1211775627.

**Reply by benallfree** [reply]
As I think about this, I recall a ticket somewhere that showed how to do data migrations in the same migration files. That might be an option here as well.

**Reply by snorresovold** [reply]
Could you post that in this thread if you find it? I'll probably try to work on that github action you mentioned, but i can't promise anything and it can take some time.

**Reply by benallfree** [reply]
https://github.com/pocketbase/pocketbase/discussions/1891
